lib/launcher.js

```diff
@@ -71,81 +60,53 @@ var Launcher = function (server, emitter, injector) {
       try {
         var browser = injector.createChild([locals], ['launcher:' + name]).get('launcher:' + name)
       } catch (e) {
-        if (e.message.indexOf('No provider for "launcher:' + name + '"') !== -1) {
-          log.error('Cannot load browser "%s": it is not registered! ' +
-            'Perhaps you are missing some plugin?', name)
+        if (e.message.indexOf(`No provider for "launcher:${name}"`) !== -1) {
+          log.error(`Cannot load browser "${name}": it is not registered! Perhaps you are missing some plugin?`)
         } else {
-          log.error('Cannot load browser "%s"!\n  ' + e.stack, name)
+          log.error(`Cannot load browser "${name}"!\n  ` + e.stack)
         }
 
         emitter.emit('load_error', 'launcher', name)
         return
       }
 
-      // TODO(vojta): remove in v1.0 (BC for old launchers)
-      if (!browser.forceKill) {
-        browser.forceKill = function () {
-          var me = this
-          return new Promise(function (resolve) {
-            me.kill(resolve)
-          })
-        }
-
-        browser.restart = function () {
-          var me = this
-          this.kill(function () {
-            me.start(url)
-          })
-        }
-      }
-
-      self.jobs.add(function (args, done) {
-        log.info('Starting browser %s', helper.isDefined(browser.displayName) ? browser.displayName : browser.name)
-
-        browser.on('browser_process_failure', function () {
-          done(browser.error)
-        })
+      this.jobs.add((args, done) => {
+        log.info(`Starting browser ${browser.displayName || browser.name}`)
 
-        browser.on('done', function () {
-          // We are not done if there was an error as first the retry takes
-          // place which we catch with `browser_process_failure` if it fails
-          if (browser.error || browser.state === browser.STATE_RESTARTING) return
+        browser.on('browser_process_failure', () => done(browser.error))
 
-          done(null, browser)
+        browser.on('done', () => {
+          if (!browser.error && browser.state !== browser.STATE_RESTARTING) {
+            done(null, browser)
+          }
         })
 
-        browser.start(url)
+        browser.start(`${protocol}//${hostname}:${port}${urlRoot}`)
       }, [])
 
-      self.jobs.run()
-      self._browsers.push(browser)
+      this.jobs.run()
+      this._browsers.push(browser)
     }
   }
 
-  this.launch = function (names, concurrency) {
+  this.launch = (names, concurrency) => {
+    lastStartTime = Date.now()
```

Comment:
To be extra careful let's put this after new Jobs as before.

Reply:
Sure, fixed